libcaf_core/caf/action.test.cpp

```diff
@@ -35,6 +38,25 @@ SCENARIO("actions wrap function calls") {
         check(uut.disposed());
       }
     }
+    WHEN("disposing an running action") {
+      auto called = false;
+      action* uut_ptr = nullptr;
+      auto uut = make_action([this, &called, &uut_ptr] {
+        called = true;
+        uut_ptr->dispose();
+        check_eq(uut_ptr->ptr()->current_state(),
+                 action::state::deferred_dispose);
+      });
+      uut_ptr = &uut;
```

Comment:
I suppose this whole `uut_ptr` setup is to have the action refer back to itself? Unless I miss something, I think it could still be a bit less verbose like this:

```suggestion
      action uut;
      uut = make_action([&called, &uut] {
        called = true;
        uut.dispose();
      });
```

Reply:
Good point. I also need the check_eq inside the lambda.